controllers/alphacontrollers/rulegroup_controller.go

```diff
@@ -124,25 +125,43 @@ func (r *RuleGroupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, nil
 	}
 
-	var notFount bool
-	var err error
-	var actualState *coralogixv1alpha1.RuleGroupStatus
+	var (
+		notFound    bool
+		err         error
+		actualState *coralogixv1alpha1.RuleGroupStatus
+	)
+
 	if id := ruleGroupCRD.Status.ID; id == nil {
 		log.V(1).Info("ruleGroup wasn't created")
-		notFount = true
+		notFound = true
 	} else if getRuleGroupResp, err := rulesGroupsClient.GetRuleGroup(ctx, &rulesgroups.GetRuleGroupRequest{GroupId: *id}); status.Code(err) == codes.NotFound {
+		// TODO: this needs more error handling, what if it returns internal error?
+
 		log.V(1).Info("ruleGroup doesn't exist in Coralogix backend")
-		notFount = true
+		notFound = true
 	} else if err == nil {
-		actualState = flattenRuleGroup(getRuleGroupResp.GetRuleGroup())
+		actualState, err = flattenRuleGroup(getRuleGroupResp.GetRuleGroup())
+		if err != nil {
+			log.Error(err, "Error mapping coralogix API response", "Name", ruleGroupCRD.Name, "Namespace", ruleGroupCRD.Namespace)
+			return ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}, err
+		}
 	}
 
-	if notFount {
+	if notFound {
 		createRuleGroupReq := ruleGroupCRD.Spec.ExtractCreateRuleGroupRequest()
-		jstr, _ := jsm.MarshalToString(createRuleGroupReq)
+		jstr, err := jsm.MarshalToString(createRuleGroupReq)
+		if err != nil {
+			log.Error(err, "Error marshalling rule group", "Name", ruleGroupCRD.Name, "Namespace", ruleGroupCRD.Namespace)
+			return ctrl.Result{RequeueAfter: defaultErrRequeuePeriod}, err
+		}
+
 		log.V(1).Info("Creating Rule-Group", "ruleGroup", jstr)
 		if createRuleGroupResp, err := rulesGroupsClient.CreateRuleGroup(ctx, createRuleGroupReq); err == nil {
-			jstr, _ := jsm.MarshalToString(createRuleGroupResp)
+			jstr, err := jsm.MarshalToString(createRuleGroupResp)
+			if err != nil {
```

Comment:
I'm not sure if we want to fail in a case of marshalling error for two reasons -
1. The marshalling is just for logs purpose and doesn't affect the actual logic of the operator.
2. In this specific case, returning an error (even though the creation succeed) will cause infinit recreation of resources in coralogix.